xdsl/parser.py

```diff
@@ -838,6 +839,17 @@ def parse_optional_mlir_vector(self,
             return VectorType.from_type_and_list(typ, dims)
         return None
 
+    def parse_optional_mlir_memref(self,
```

Comment:
Nice, could you also add the unranked memref? This should be implemented like the `UnrankedTensorType`.